### PR TITLE
Android comment enhancement

### PR DESCRIPTION
**Pull request by mmutas**
screenshot_20171205-032746

Changes proposed in this pull request:

- User name and user picture added to the comments.
- Comment writing section improved.

### CODE REVIEW

File: Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/adapter/CommentRow.java [hunk 1]
```diff
@@ -34,14 +45,34 @@ public void setText(String text) {
...
+
     public CommentRow() {
 
     }
 
-    public CommentRow(String name, String date, String text) {
+    public CommentRow(String name, String date, String text, String avatar) {
         this.name = name;
+
+        String[] parts = date.split("[.]");
```
Comment (eozd) [inline Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/adapter/CommentRow.java]: I believe the functionality of this code piece is the same as ```setDate```. Can you call the function to reduce code duplication

File: Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/adapter/CommentAdapter.java [hunk 2]
```diff
@@ -42,21 +57,63 @@ public View getView(int pos,View convertView, ViewGroup parent) {
...
         return convertView;
     }
 
     @Override
     public int getCount() {
         return this.items.size();
     }
+
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    public String getDateString(String date) {
```
Comment (eozd) [inline Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/adapter/CommentAdapter.java:85]: The functionality of this function seems to be complex. Can you write unit tests to ensure that it works properly?

File: Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/adapter/CommentRow.java [hunk 3]
```diff
@@ -23,7 +24,17 @@ public String getDate() {
     }
 
     public void setDate(String date) {
+
```
Comment (eozd) [inline Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/adapter/CommentRow.java:27]: The functionality of this function seems to be complex. Can you write unit tests to ensure that it works properly?

**Review by eozd — CHANGES_REQUESTED** [review]

File: Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/adapter/CommentRow.java (hunk 1 above)
Comment (mmutas) [reply Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/adapter/CommentRow.java]: done.

File: Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/adapter/CommentRow.java (hunk 3 above)
Comment (mmutas) [reply Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/adapter/CommentRow.java:27]: will be done later

File: Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/adapter/CommentAdapter.java (hunk 2 above)
Comment (mmutas) [reply Atlas/android/app/src/main/java/com/bounswe2017/group10/atlas/adapter/CommentAdapter.java:85]: will be done later

**Review by eozd — APPROVED** [review]

**Pull request merged**